src/dbwrapper.h

```diff
@@ -207,12 +222,17 @@ class CDBWrapper
     CDBWrapper(const fs::path& path, size_t nCacheSize, bool fMemory = false, bool fWipe = false);
     ~CDBWrapper();
 
-    template <typename K, typename V>
-    bool Read(const K& key, V& value) const
+    template <typename K>
+    bool ReadDataStream(const K& key, CDataStream& ssValue) const
     {
         CDataStream ssKey(SER_DISK, CLIENT_VERSION);
         ssKey.reserve(DBWRAPPER_PREALLOC_KEY_SIZE);
         ssKey << key;
+        return ReadDataStream(ssKey, ssValue);
+    }
+
+    bool ReadDataStream(const CDataStream& ssKey, CDataStream& ssValue) const
```

Comment:
I'm not sure how this is working.

So, i'm seeing the following flow: `Read(const K& key, V& value)` calls `Read(const CDataStream& ssKey, V& value)` which calls `ReadDataStream(const CDataStream& ssKey, CDataStream& ssValue`.

Inside the last function the `ssValue` of `ReadDataStream` is never assigned nor used inside the function. Line 227 is creating a new object instead. So when it get back to the first `Read`, `ssValue` has no value, so `value` is not assigned.

Reply:
ok, found why it's working. You solved it in a later commit f259774db579dd03b22307d3b069a22c10975c70 .